### PR TITLE
Implement Registration and Email Confirmation Services

### PR DESCRIPTION
**Pull request by alaahewwari**

### CODE REVIEW

File: ECommerceAPI/Constants/Email/EmailUrlConstants.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿
+namespace ECommerceAPI.Constants.Email
+{
+    public class EmailUrlConstants
+    {
+        public static readonly string ConfirmationImageUrl = "https://th.bing.com/th/id/OIP.i_nEQw6xesPjDvPsPgqcRwHaHa?rs=1&pid=ImgDetMain";
```
Comment (MohammadDiab) [inline ECommerceAPI/Constants/Email/EmailUrlConstants.cs:6]: image should be in project assets

File: ECommerceAPI/Configurations/EndpointsConfiguration.cs [hunk 2]
```diff
@@ -4,6 +4,11 @@ namespace ECommerceAPI.Configurations
 {
     public static class EndpointsConfiguration 
     {
+        public static void MapAuthenticationEndpoints(this WebApplication app)
+        {
+            app.MapPost("/api/auth/registration", AuthenticationEndpoints.Register);
+        app.MapGet("/api/auth/email-confirmation", AuthenticationEndpoints.ConfirmEmail);
+        }
```
Comment (MohammadDiab) [inline ECommerceAPI/Configurations/EndpointsConfiguration.cs:11]: please fix spacing

File: ECommerceAPI/ECommerceAPI.csproj [hunk 3]
```diff
@@ -9,15 +9,15 @@
   <ItemGroup>
     <PackageReference Include="AutoMapper" Version="13.0.1" />
     <PackageReference Include="ErrorOr" Version="2.0.1" />
-    <PackageReference Include="Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore" Version="8.0.4" />
+    <PackageReference Include="MailKit" Version="4.6.0" />
```
Comment (MohammadDiab) [inline ECommerceAPI/ECommerceAPI.csproj:12]: why you need? you can use third party for email services like SendGrid

File: ECommerceAPI/ECommerceAPI.csproj [hunk 4]
```diff
@@ -9,15 +9,15 @@
...
     <PackageReference Include="Microsoft.EntityFrameworkCore" Version="8.0.4" />
     <PackageReference Include="Microsoft.EntityFrameworkCore.InMemory" Version="8.0.4" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="8.0.4" />
+    <PackageReference Include="Microsoft.AspNetCore.Identity.EntityFrameworkCore" Version="8.0.5" />
     <PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="8.0.4">
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
-    <PackageReference Include="Swashbuckle.AspNetCore" Version="6.4.0" />
+    <PackageReference Include="MimeKit" Version="4.6.0" />
```
Comment (MohammadDiab) [inline ECommerceAPI/ECommerceAPI.csproj:20]: and this also, why you needed

File: ECommerceAPI/Endpoints/AuthenticationEndpoints.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+using ECommerceAPI.Dtos.Authentication.Requests;
+using ECommerceAPI.Services.AuthenticationServices.Interfaces;
+
+namespace ECommerceAPI.Endpoints
+{
+    public class AuthenticationEndpoints 
+    {
+        public static async Task<IResult> Register(IAuthenticationService authenticationService, RegistrationRequestDto registerRequest)
+        {
+            var baseUrl = "https://localhost:7131/api/auth/email-confirmation";
```
Comment (MohammadDiab) [inline ECommerceAPI/Endpoints/AuthenticationEndpoints.cs:11]: I prefer to build URL more dynamic not static string

File: ECommerceAPI/ErrorHandling/IdentityErrors.cs [hunk 6]
```diff
@@ -0,0 +1,23 @@
+﻿using ErrorOr;
+
+namespace ECommerceAPI.ErrorHandling
+{
+    public static class IdentityErrors
+    {
```
Comment (MohammadDiab) [inline ECommerceAPI/ErrorHandling/IdentityErrors.cs:6]: please fix space.

File: ECommerceAPI/Infrastructure/Configurations/EmailConfiguration.cs [hunk 7]
```diff
@@ -0,0 +1,11 @@
+﻿namespace ECommerceAPI.Infrastructure.Configurations
+{
+    public class EmailConfiguration
+    {
+        public string From { get; set; } = null!;
+        public string SmtpServer { get; set; } = null!;
+        public int Port { get; set; }
+        public string UserName { get; set; } = null!;
+        public string Password { get; set; } = null!;
```
Comment (MohammadDiab) [inline ECommerceAPI/Infrastructure/Configurations/EmailConfiguration.cs:9]: string by default is null

File: ECommerceAPI/Services/AuthenticationServices/Implementations/AuthenticationService.cs [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+using Microsoft.IdentityModel.Tokens;
+
+namespace ECommerceAPI.Services.AuthenticationServices.Implementations
+{
+    public class AuthenticationService(
+        IIdentityService identityService,
+        IEmailService emailService)
+        : IAuthenticationService
+    {
+        public async Task<ErrorOr<SuccessResponse>> RegisterAsync(RegistrationRequestDto request,string url)
```
Comment (MohammadDiab) [inline ECommerceAPI/Services/AuthenticationServices/Implementations/AuthenticationService.cs:19]: please fix spacing

File: ECommerceAPI/ECommerceAPI.csproj (hunk 3 above)
Comment (alaahewwari) [reply ECommerceAPI/ECommerceAPI.csproj:12]: For email messages, 
Today I tried to learn how to use SendGrid but I found some problems, such as this message
 "We are unable to log you in at this time, please check your email for further information."
 that appears when I want to log into the account
I'll try again later

File: ECommerceAPI/Infrastructure/Configurations/EmailConfiguration.cs (hunk 7 above)
Comment (alaahewwari) [reply ECommerceAPI/Infrastructure/Configurations/EmailConfiguration.cs]: You're right, but I used the null-forgiving operator (null!)

**Review by MohammadDiab — APPROVED** [review]

**Pull request merged**